test/inductor/test_distributed_patterns.py

```diff
@@ -437,12 +450,28 @@ def test_fake_distributed_inductor(self):
 
         m2, inp2 = init_fake_distributed()
         m2 = torch.compile(m2, fullgraph=True)
-        with compiled_autograd.enable(torch.compile(fullgraph=True)):
-            out2 = steps(m2, inp2)
+        # The forward runs successfully, but functionalizing the backward errors today.
+        # See bullet (8) of the description at https://github.com/pytorch/pytorch/pull/120971 for more  details.
+        # TLDR: we see the parameter as two separate inputs in the bw graph:
+        # (a) one from a saved activation (potential the weight.t())
+        # (b) one from the pre-backward hook closing over the parameter
+        # The second is what actually sees the resize_() / copy in,
+        # while the first is used in the actual compute.
+        # The easiest way to handle this would be to figure out how to de-duplicate the parameter
+        # before we functionalize the backward.
+        # This can be done either as:
+        # (1) a custom FX pass (Will F. has pass for it, although it is not guaranteed to be safe)
+        # (2) Detecting when input aliases an safely be deduplicated and regenerated inside of the graph.
+        #     This is likely safer but will be a reasonable amount of work
+        with self.assertRaisesRegex(
+            RuntimeError, "requiring a storage size of 800 are out of bounds"
+        ):
+            with compiled_autograd.enable(torch.compile(fullgraph=True)):
+                out2 = steps(m2, inp2)
```

Comment:
Can you create a copy of this test with `@patch(f"{__name__}.RESIZE", False)` to ensure you aren't breaking the existing functionality in inductor?   I'm pretty sure removing the bind nn parameter op from inductor and adding the `.detatch` is a regression in the old version of this test.

Reply:
@jansel I added one locally, and it actually passes (will push it later today). I checked the inductor logs, and I think it's because:

(1) inductor generates an implicit fallback for `aten.set_` (the op the in the graph that replaces _bind_nn_parameter)

(2) the logic for the implicit fallback looks like it's pretty similar to the custom lowering for `_bind_nn_parameter`, except that `_bind_nn_parameter` properly marks the node as mutating (so the implicit fallback could probably be reordered incorrectly, but isn't in this test).

Reply:
Let me know which option you'd prefer:

(1) Add a new lowering for `set_()` in inductor, that looks pretty similar to the `_bind_nn_parameter` lowering today (effectively replacing `_bind_nn_parameter` with its more generic version)

(2) Stop using `set_()` completely, just mark `_bind_nn_parameter` as mutable and plumb it through AOTAutograd similar to how I'm plumbing `set_`.

I think using `set_()` feels a bit better since it's a more generic op (and for whichever one we pick, we need to special-case it in AOTAutograd), but either seems ok to me

Reply:
The implicit fallback for `aten.set_` will have the wrong mutation/aliasing annotations in inductor which will cause issues.

Let's do option (1)